Networking/Networking/Model/BlazeCampaign.swift

```diff
@@ -130,4 +139,16 @@ private extension BlazeCampaign {
     enum DecodingError: Error {
         case missingSiteID
     }
+
+    /// Extracts the product ID from the `target_urn` response.
+    /// The response looks like the following: `urn:wpcom:post:1:134`
+    /// The product ID is the last number following the colon in the response (`134`).
+    /// If the product ID cannot be found, zero will be returned, to ensure that the entire Campaign response is not rejected.
+    /// Product ID is currently used only for showing or hiding the Blaze button on product details, and its absence is not a
+    /// dealbreaker.
```

Comment:
to reviewer: wdyt about this reasoning?

Reply:
I'd prefer keeping `productID` optional as we're not sure if the 0 id can point to any actual product. WDYT @selanthiraiyan?

Reply:
Not sure if it helps to answer but in the `wp_posts` table, ID is a primary key so it can't be zero:

https://codex.wordpress.org/Database_Description#Table:_wp_posts

Reply:
I updated things to make ProductID optional 👍🏼